h2quic/client.go

```diff
@@ -147,14 +142,16 @@ func (c *client) Do(req *http.Request) (*http.Response, error) {
 		return nil, fmt.Errorf("h2quic Client BUG: RoundTrip called for the wrong client (expected %s, got %s)", c.hostname, req.Host)
 	}
 
-	hasBody := (req.Body != nil)
+	c.dialOnce.Do(func() {
+		c.handshakeErr = c.dial()
```

Comment:
No, because the Do is only executed once, but the error has to be returned from all calls to Roundtrip.